src/modules/ce/components/ReferralStepSummary.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useMemo } from 'react';
+import { useNavigate } from 'react-router-dom';
+import ButtonLink, { ButtonLinkProps } from '@/components/elements/ButtonLink';
+import CommonCard from '@/components/elements/CommonCard';
+import LoadingButton from '@/components/elements/LoadingButton';
+import { CompletedIcon, GoToIcon } from '@/components/elements/SemanticIcons';
+import useSafeParams from '@/hooks/useSafeParams';
+import { useReferralContext } from '@/modules/ce/components/ReferralNav';
+import ReferralStepAssignee from '@/modules/ce/components/ReferralStepAssignee';
+import { ProjectDashboardRoutes } from '@/routes/routes';
+import {
+  CeReferralStepStatus,
+  CeReferralStepSummaryFieldsFragment,
+  useStartCeReferralStepMutation,
+} from '@/types/gqlTypes';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+interface Props {
+  step: CeReferralStepSummaryFieldsFragment;
+}
+const ReferralStepSummary: React.FC<Props> = ({ step }) => {
```

Comment:
Noting that this component has a mix of state/data/mutation logic and presentation logic.  Since this card has complex styling needs (varying colors/icon based on state) it seems like a good candidate to pull out into a stateless presentational component.  That would make this component easier to reason about.

```jsx
// container component
const ReferralStepCard: React.FC<Props> = ({ step }) => {
    // mutation here, define actions
   return (
       //presentational component
      <ReferralStepCardInner status={referral.status} action={action}>
        {/* card content */}
      </ReferralStepCardInner>
    )
});
```